specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobPrivateEndpoints.json

```diff
@@ -0,0 +1,337 @@
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "The specified private endpoint does not exist."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Delete a private endpoint": {
+            "$ref": "./examples/DeleteJobPrivateEndpoint.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "JobPrivateEndpoint": {
+      "description": "A job agent private endpoint.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/JobPrivateEndpointProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "JobPrivateEndpointListResult": {
+      "description": "A list of job agent private endpoints.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobPrivateEndpoint"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "JobPrivateEndpointProperties": {
+      "description": "Properties of job agent private endpoint.",
+      "required": [
+        "targetServerAzureResourceId"
+      ],
+      "type": "object",
+      "properties": {
+        "targetServerAzureResourceId": {
+          "description": "ARM resource id of the server the private endpoint will target.",
+          "type": "string",
+          "format": "arm-id",
+          "x-ms-arm-id-details": {
+            "allowedResources": [
+              {
+                "type": "Microsoft.Sql/servers"
+              }
+            ]
+          }
+        },
+        "privateEndpointId": {
+          "description": "Private endpoint id of the private endpoint.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ServerNameParameter": {
```

Comment:
The resource name parameter 'serverName' should be defined with a 'pattern' restriction.

Reply:
I could not find this pattern restriction in any of our SQL swagger files. This is not something our existing generator supports. You can see that the parent resource currently in production 'JobAgents.json' does not have any pattern restrictions. Please feel free to point me to any SQL swagger file that has this pattern restriction if I am mistaken. 

// parent resource currently live in production. 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/JobAgents.json